docs/src/pages/components/data-grid/components/components.md

```diff
@@ -32,20 +105,32 @@ function CustomToolbar() {
 
 {{"demo": "pages/components/data-grid/components/CustomToolbarGrid.js", "bg": "inline"}}
 
-## Loading overlay
+### Header
 
-By default, the loading overlay displays a circular progress.
-This demo replaces it with a linear progress.
+The default header component is the **parent** component of the **toolbar** and of the **filter and columns panel**.
```

Comment:
Could we remove the GridHeader customization slot and not document it? It seems to be a foot-gun with no advantages over using the toolbar slot.